griptape/config/openai_structure_config.py

```diff
@@ -0,0 +1,45 @@
+from attrs import define, field, Factory
+from griptape.drivers import BasePromptDriver, OpenAiChatPromptDriver, LocalVectorStoreDriver, OpenAiEmbeddingDriver
+from griptape.config import (
+    BaseStructureConfig,
+    StructureTaskMemoryConfig,
+    StructureTaskMemoryQueryEngineConfig,
+    StructureTaskMemoryExtractionEngineConfig,
+    StructureTaskMemorySummaryEngineConfig,
+    StructureTaskMemoryExtractionEngineJsonConfig,
+    StructureTaskMemoryExtractionEngineCsvConfig,
+)
+
+
+@define(kw_only=True)
+class OpenAiStructureConfig(BaseStructureConfig):
+    prompt_driver: BasePromptDriver = field(
+        default=Factory(lambda: OpenAiChatPromptDriver(model="gpt-4", stream=False)),
+        kw_only=True,
+        metadata={"save": True},
+    )
+    task_memory: StructureTaskMemoryConfig = field(
+        default=Factory(
+            lambda: StructureTaskMemoryConfig(
+                query_engine=StructureTaskMemoryQueryEngineConfig(
+                    prompt_driver=OpenAiChatPromptDriver(model="gpt-3.5-turbo-16k"),
+                    vector_store_driver=LocalVectorStoreDriver(
+                        embedding_driver=OpenAiEmbeddingDriver(model="text-embedding-ada-002")
+                    ),
+                ),
+                extraction_engine=StructureTaskMemoryExtractionEngineConfig(
+                    csv=StructureTaskMemoryExtractionEngineCsvConfig(
+                        prompt_driver=OpenAiChatPromptDriver(model="gpt-3.5-turbo-16k")
```

Comment:
I'd like to get input from the folks who have been working on Griptape Cloud how a text-based config file could be hitched up to BaseStructureConfig in order to make this truly data-driven

Reply:
this config file can be set on the App level and loaded in when the app runs. it could also be set on the Deployment level, i think there is pros and cons to both.